cpu/stm32/stmclk/stmclk_l0l1.c

```diff
@@ -22,67 +22,166 @@
+#define REG_CIR     (RCC->CICR)
+#endif
+
+/* configuration of flash access cycles */
+#define CLOCK_FLASH_LATENCY         (FLASH_ACR_LATENCY)
+
+/* Configure the prescalers */
+#define CLOCK_AHB_DIV               (RCC_CFGR_HPRE_DIV1)  /* HCLK = SYSCLK */
+
+#if CONFIG_CLOCK_APB1_DIV == 1
+#define CLOCK_APB1_DIV              (RCC_CFGR_PPRE1_DIV1)
+#elif CONFIG_CLOCK_APB1_DIV == 2
+#define CLOCK_APB1_DIV              (RCC_CFGR_PPRE1_DIV2)
+#elif CONFIG_CLOCK_APB1_DIV == 4
+#define CLOCK_APB1_DIV              (RCC_CFGR_PPRE1_DIV4)
+#elif CONFIG_CLOCK_APB1_DIV == 8
+#define CLOCK_APB1_DIV              (RCC_CFGR_PPRE1_DIV8)
+#elif CONFIG_CLOCK_APB1_DIV == 16
+#define CLOCK_APB1_DIV              (RCC_CFGR_PPRE1_DIV16)
+#endif
+
+#if CONFIG_CLOCK_APB2_DIV == 1
+#define CLOCK_APB2_DIV              (RCC_CFGR_PPRE2_DIV1)
+#elif CONFIG_CLOCK_APB2_DIV == 2
+#define CLOCK_APB2_DIV              (RCC_CFGR_PPRE2_DIV2)
+#elif CONFIG_CLOCK_APB2_DIV == 4
+#define CLOCK_APB2_DIV              (RCC_CFGR_PPRE2_DIV4)
+#elif CONFIG_CLOCK_APB2_DIV == 8
+#define CLOCK_APB2_DIV              (RCC_CFGR_PPRE2_DIV8)
+#elif CONFIG_CLOCK_APB2_DIV == 16
+#define CLOCK_APB2_DIV              (RCC_CFGR_PPRE2_DIV16)
+#endif
 
 /* Check the source to be used for the PLL */
-#if defined(CLOCK_HSI) && defined(CLOCK_HSE)
-#error "Only provide one of two CLOCK_HSI/CLOCK_HSE"
-#elif CLOCK_HSI
-#define CLOCK_CR_SOURCE            RCC_CR_HSION
-#define CLOCK_CR_SOURCE_RDY        RCC_CR_HSIRDY
-#define CLOCK_PLL_SOURCE           RCC_CFGR_PLLSRC_HSI
-#elif CLOCK_HSE
-#define CLOCK_CR_SOURCE            RCC_CR_HSEON
-#define CLOCK_CR_SOURCE_RDY        RCC_CR_HSERDY
-#define CLOCK_PLL_SOURCE           RCC_CFGR_PLLSRC_HSE
+#if IS_ACTIVE(CONFIG_BOARD_HAS_HSE)
+#define CLOCK_PLL_SOURCE            (RCC_CFGR_PLLSRC_HSE)
+#else /* Use HSI as PLL input */
+#define CLOCK_PLL_SOURCE            (RCC_CFGR_PLLSRC_HSI)
+#endif
+
+#if CONFIG_CLOCK_PLL_DIV == 2
+#define CLOCK_PLL_DIV               (RCC_CFGR_PLLDIV2)
+#elif CONFIG_CLOCK_PLL_DIV == 3
+#define CLOCK_PLL_DIV               (RCC_CFGR_PLLDIV3)
+#elif CONFIG_CLOCK_PLL_DIV == 4
+#define CLOCK_PLL_DIV               (RCC_CFGR_PLLDIV4)
+#endif
+
+#if CONFIG_CLOCK_PLL_MUL == 3
+#define CLOCK_PLL_MUL               (RCC_CFGR_PLLMUL3)
+#elif CONFIG_CLOCK_PLL_MUL == 4
+#define CLOCK_PLL_MUL               (RCC_CFGR_PLLMUL4)
+#elif CONFIG_CLOCK_PLL_MUL == 6
+#define CLOCK_PLL_MUL               (RCC_CFGR_PLLMUL6)
+#elif CONFIG_CLOCK_PLL_MUL == 8
+#define CLOCK_PLL_MUL               (RCC_CFGR_PLLMUL8)
+#elif CONFIG_CLOCK_PLL_MUL == 12
+#define CLOCK_PLL_MUL               (RCC_CFGR_PLLMUL12)
+#elif CONFIG_CLOCK_PLL_MUL == 16
+#define CLOCK_PLL_MUL               (RCC_CFGR_PLLMUL16)
+#elif CONFIG_CLOCK_PLL_MUL == 24
+#define CLOCK_PLL_MUL               (RCC_CFGR_PLLMUL24)
+#elif CONFIG_CLOCK_PLL_MUL == 32
+#define CLOCK_PLL_MUL               (RCC_CFGR_PLLMUL32)
+#elif CONFIG_CLOCK_PLL_MUL == 48
+#define CLOCK_PLL_MUL               (RCC_CFGR_PLLMUL48)
+#endif
```

Comment:
I changed these values through the command line but I did not get an error message when I set an invalid value, maybe at the end an error message? To not only have a compilation error:

`CFLAGS="-DCONFIG_CLOCK_PLL_DIV=8"`

```
/home/francisco/workspace/RIOT3/cpu/stm32/stmclk/stmclk_l0l1.c:215:42: error: 'CLOCK_PLL_DIV' undeclared (first use in this function); did you mean 'CLOCK_PLL_MUL'?
         RCC->CFGR |= (CLOCK_PLL_SOURCE | CLOCK_PLL_DIV | CLOCK_PLL_MUL);
                                          ^~~~~~~~~~~~~
                                          CLOCK_PLL_MUL
```

Reply:
I added an else case that raises an error. So the build will fail if incorrect PLL parameters are give.